library/objective-c/EnvoyEngineImpl.m

```diff
@@ -365,12 +365,26 @@ static envoy_data ios_get_string(const void *context) {
   return toManagedNativeString(accessor.getEnvoyString());
 }
 
+static void ios_track_event(envoy_map map, const void *context) {
+  // This code block runs inside the Envoy event loop. Therefore, an explicit autoreleasepool block
+  // is necessary to act as a breaker for any Objective-C allocation that happens.
+  @autoreleasepool {
+    EnvoyEventTracker *eventTracker = (__bridge EnvoyEventTracker *)context;
+    // release memory even if the tracking closure is not provided.
+    EnvoyEvent *platformMap = to_ios_map(map);
```

Comment:
Why not move this into the `if` statement to avoid creating this if there is no `track` closure?

Reply:
See the comment above, `to_ios_map` method call releases the memory so we want to ensure that it's always called.